tmt/hardware.py

```diff
@@ -731,12 +731,46 @@ def _cast_number(raw_value: Any) -> int:
             name,
             raw_value,
             as_quantity=False,
-            as_cast=_cast_number,
+            as_cast=_cast_int,
             original_constraint=original_constraint,
             allowed_operators=allowed_operators
             )
 
 
+class NumberConstraint(Constraint[float]):
+    """ A constraint representing a dimension-less float number """
```

Comment:
`IntegerConstraint` remains dimension-less, but `NumberConstraint` isn't, we already use it for a constraint with a default unit :)